From d5f72e9b8d857e79a17dc4d10b2021a9110e49fe Mon Sep 17 00:00:00 2001
From: Saurav Babu <saurav.babu@samsung.com>
Date: Tue, 1 Dec 2015 17:39:33 +0530
Subject: dhcp: Fixed memory leak

Memory allocated should be freed when apply_lease_available_on_network()
fails.
---
 src/dhcp.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/dhcp.c b/src/dhcp.c
index 9a74362..e4497fc 100644
--- a/src/dhcp.c
+++ b/src/dhcp.c
@@ -462,11 +462,12 @@ static void lease_available_cb(GDHCPClient *dhcp_client, gpointer user_data)
 	}
 
 	if (!apply_lease_available_on_network(dhcp_client, dhcp))
-		return;
+		goto done;
 
 	if (ip_change)
 		dhcp_valid(dhcp);
 
+done:
 	g_free(address);
 	g_free(netmask);
 	g_free(gateway);
-- 
cgit v0.11.2

